### PR TITLE
Ncsdk 14460 event manager icmsg

### PR DESCRIPTION
**Pull request by rakons**
Add icmsg backend configuration to event_manager_proxy sample.

### CODE REVIEW

**Comment by rakons** [comment]
Waiting for #7093.

**Comment by rakons** [comment]
Note: I have added the missing information about event_manager_proxy library to the changelog.

File: samples/event_manager_proxy/README.rst [hunk 1]
```diff
@@ -74,6 +74,15 @@ For example, you can do this from the command line using west by completing the
...
 
+Selecting ICMSG backend
+=======================
+
+The sample is provided with ICMSG backend configuration. To use it, specify the ``-DCONF_FILE=prj_icmsg.conf`` parameter along with the build command:
+
+  .. code-block:: console
+
+     west build -p -b nrf5340dk_nrf5340_cpuapp -- -DCONF_FILE=prj_icmsg.conf
+
```
Comment (divipillai) [inline samples/event_manager_proxy/README.rst:85]: The whole steps should be moved after step 63 as we are building the sample at that step. The section heading should be removed.
```suggestion
 To provide the ICMSG backend configuration, specify the ``-DCONF_FILE=prj_icmsg.conf`` parameter along with the build command:

  .. code-bloc:: console

     west build -p -b nrf5340dk_nrf5340_cpuapp -- -DCONF_FILE=prj_icmsg.conf

```

**Review by pdunaj — APPROVED** [review]

File: samples/event_manager_proxy/boards/nf5340dk_nrf5340_cpuapp.conf [hunk 2]
```diff
@@ -0,0 +1 @@
+CONFIG_BOARD_ENABLE_CPUNET=y
```
Comment (emob-nordic) [inline samples/event_manager_proxy/boards/nf5340dk_nrf5340_cpuapp.conf:1]: Filename is wrong. You are missing `r`.

Comment (rakons) [reply samples/event_manager_proxy/boards/nf5340dk_nrf5340_cpuapp.conf:1]: Right. There was second error before that hides this one during tests.

File: doc/nrf/releases/release-notes-changelog.rst [hunk 3]
```diff
@@ -257,7 +257,10 @@ Zigbee samples
 Other Samples
 -------------
 
-* Added the :ref:`ipc_service_sample` sample, showing throughtput of IPC service with available backends.
+* Added:
+  * :ref:`ipc_service_sample` sample, showing throughput of IPC service with available backends. 
+  * :ref:`event_manager_proxy_sample` sample, which demonstrates the usage of :ref:`event_manager_proxy`.
+``
```
Comment (greg-fer) [inline doc/nrf/releases/release-notes-changelog.rst:263]:
```suggestion
* Added:

  * :ref:`ipc_service_sample` sample, which demonstrates throughput of the IPC service with available backends. 
  * :ref:`event_manager_proxy_sample` sample, which demonstrates the usage of :ref:`event_manager_proxy`.
```

File: samples/event_manager_proxy/README.rst [hunk 4]
```diff
@@ -74,6 +74,12 @@ For example, you can do this from the command line using west by completing the
       cd build/event_manager_proxy_remote-prefix/src/event_manager_proxy_remote-build/
       west flash
 
+ To provide the ICMSG backend configuration, specify the ``-DCONF_FILE=prj_icmsg.conf`` parameter along with the build command:
+
+  .. code-bloc:: console
+
+     west build -p -b nrf5340dk_nrf5340_cpuapp -- -DCONF_FILE=prj_icmsg.conf
+
```
Comment (greg-fer) [inline samples/event_manager_proxy/README.rst:82]:
```suggestion
ICMSG backend configuration
 To provide the ICMSG backend configuration, specify the ``-DCONF_FILE=prj_icmsg.conf`` parameter along with the build command:

  .. code-bloc:: console

     west build -p -b nrf5340dk_nrf5340_cpuapp -- -DCONF_FILE=prj_icmsg.conf

```
One concern here: this is the only place where we mention ICMSG configuration. For this reason, I would rather see these lines in a "Configuration" section with additional information about the ICMSG configuration, even just one line of context.

**Review by greg-fer — CHANGES_REQUESTED** [review]

File: samples/event_manager_proxy/boards/nrf5340dk_nrf5340_cpuapp_icmsg.overlay [hunk 5]
```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
```
Comment (zycz) [inline samples/event_manager_proxy/boards/nrf5340dk_nrf5340_cpuapp_icmsg.overlay:4]: Change license. Also in other places.

**Review by zycz — APPROVED** [review]

**Comment by rakons** [comment]
Changed to child image

**Review by greg-fer — APPROVED** [review]

**Comment by rakons** [comment]
Changes to the build system to remove any direct reference to board names from the main CMakeFile

File: samples/event_manager_proxy/boards/nrf5340dk_nrf5340_cpuapp_icmsg.overlay (hunk 5 above)
Comment (rakons) [reply samples/event_manager_proxy/boards/nrf5340dk_nrf5340_cpuapp_icmsg.overlay]: Done.

**Pull request merged**